lapce-data/src/config.rs

```diff
@@ -538,30 +538,91 @@ impl Config {
         self.themes.style_color(name)
     }
 
+    /// Calculate the width of the character "W" (being the widest character)
+    /// in the editor's current font family at the specified font size.
     pub fn char_width(&self, text: &mut PietText, font_size: f64) -> f64 {
-        let text_layout = text
-            .new_text_layout("W")
-            .font(self.editor.font_family(), font_size)
-            .build()
-            .unwrap();
-        text_layout.size().width
+        Self::editor_text_size_internal(
+            self.editor.font_family(),
+            font_size,
+            text,
+            "W",
+        )
+        .width
     }
 
-    pub fn editor_text_width(&self, text: &mut PietText, c: &str) -> f64 {
-        let text_layout = text
-            .new_text_layout(c.to_string())
-            .font(self.editor.font_family(), self.editor.font_size as f64)
-            .build()
-            .unwrap();
-        text_layout.size().width
+    /// Calculate the width of the character "W" (being the widest character)
+    /// in the editor's current font family and current font size.
+    pub fn editor_char_width(&self, text: &mut PietText) -> f64 {
+        Self::editor_text_size_internal(
+            self.editor.font_family(),
+            self.editor.font_size as f64,
+            text,
+            "W",
+        )
+        .width
+    }
+
+    /// Calculate the width of `text_to_measure` in the editor's current font family and font size.
+    pub fn editor_text_width(
+        &self,
+        text: &mut PietText,
+        text_to_measure: &str,
+    ) -> f64 {
+        Self::editor_text_size_internal(
```

Comment:
Subjective: This could call `editor_text_size` directly instead of duplicating the font size arg.